py/torch_tensorrt/dynamo/lowering/passes/lower_linear.py

```diff
@@ -0,0 +1,67 @@
+import logging
+from typing import Callable, Sequence, Tuple
+
+import torch
+from torch_tensorrt.dynamo.lowering.passes.pass_utils import (
+    clean_up_graph_after_modifications,
+    get_tensor_placeholders,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def lower_linear(
+    gm: torch.fx.GraphModule, sample_inputs: Sequence[torch.Tensor]
+) -> torch.fx.GraphModule:
+    """Replace aten.linear with an equivalent implementation which can be easily converted to TRT"""
+    orig, replacement = linear_replacement()
+
+    if torch.fx.subgraph_rewriter.replace_pattern(gm, orig, replacement):
+        gm = clean_up_graph_after_modifications(gm)
+        logger.debug(f"Graph after lowering linear:\n{gm.graph}")
+
+    return gm
+
+
+def linear_replacement() -> (
+    Tuple[
+        torch.fx.GraphModule,
+        Callable[[torch.Tensor, torch.Tensor, torch.Tensor], torch.Tensor],
+    ]
+):
+    """Constructs the original and replacement functions for linear"""
+
+    # Empty boilerplate function taking in three Tensors and returning one
+    def boilerplate(
+        input: torch.Tensor, weight: torch.Tensor, bias: torch.Tensor
+    ) -> torch.Tensor:
+        ...
+
+    # Trace boilerplate function and extract placeholder and output nodes
+    orig = torch.fx.symbolic_trace(boilerplate)
+    input, weight, bias = get_tensor_placeholders(orig)
+    output = [node for node in orig.graph.nodes if node.op == "output"][0]
+
+    with orig.graph.inserting_before(output):
+        W_T = orig.graph.call_function(
+            torch.ops.aten.permute.default,
+            args=(weight, [1, 0]),
+        )
+        out = orig.graph.call_function(
+            torch.ops.aten.addmm.default,
+            args=(bias, input, W_T),
+        )
+
+    # Assign the output of the graph to be the single getitem output
+    output.args = (out,)
+
+    orig.graph.lint()
+    orig.recompile()
```

Comment:
The `linear` operator may not necessarily have the same restrictions as the attention operator did, so it may not be necessary to build the fx graph from scratch. For instance, this whole block could instead be:
```python
    def orig(
        input: torch.Tensor, weight: torch.Tensor, bias: torch.Tensor
    ) -> torch.Tensor:
        W_T = torch.ops.aten.permute.default(weight, [1, 0])
        out = torch.ops.aten.addmm.default(bias, input, W_T)
        return out
```
The above would need verification to validate functionality.

Reply:
Thanks for the details. I'm thinking why we need these restrictions for `attention` but not for `linear`. Is there anything different in `attention`? Can we just define the `orig` like:
```python
def orig(
        q: torch.Tensor, k: torch.Tensor, v: torch.Tensor
    ) -> torch.Tensor:
        outputs = torch.ops.aten._scaled_dot_product_efficient_attention.default(q, k, v, None, False)
        out = operator.getitem(outputs, 0)
        return out
```

Reply:
I just tried to use the `orig` function above, and it also works for `attention`.

Reply:
Ok that's great - the above is definitely cleaner and the preferred usage. I think I had overlooked it or was working on a Torch nightly where that wasn't the case. If you could add the change to the attention pass as well for this PR, that would be much appreciated.